pandas/core/arrays/categorical.py

```diff
@@ -331,7 +331,7 @@ def __init__(self, values, categories=None, ordered=None, dtype=None,
         # sanitize input
         if is_categorical_dtype(values):
             if dtype.categories is None:
-                dtype = CategoricalDtype(values.categories, dtype.ordered)
+                dtype = CategoricalDtype(values.categories, dtype._ordered)
```

Comment:
we really need to user the internal one here? (and all others); I *really* don't want to expose that even to our code.

Reply:
See the overview that Jeremy gave above (https://github.com/pandas-dev/pandas/pull/26403#issuecomment-507351745) and this comment for more details on why `_ordered` was needed: https://github.com/pandas-dev/pandas/pull/26403#discussion_r299144624 
That explains clearly the context on why this was added, and I am fine with it.

Reply:
ok that's fine then, @jschendel can you create an issue for this so we don't forget at removal time.

Reply:
sure, back at work today so will create the issue later on tonight